sdk/nanotdf.go

```diff
@@ -841,9 +857,20 @@ func (s SDK) ReadNanoTDF(writer io.Writer, reader io.ReadSeeker) (uint32, error)
 }
 
 // getECPublicKey - Contact the specified KAS and get its public key
-func getECPublicKey(kasURL string, opts ...grpc.DialOption) (string, error) {
+func getECPublicKey(kasURL string, ecMode ocrypto.ECCMode, opts ...grpc.DialOption) (string, error) {
 	req := kas.PublicKeyRequest{}
-	req.Algorithm = "ec:secp256r1"
+	switch ecMode {
+	case ocrypto.ECCModeSecp256r1:
+		req.Algorithm = "ec:secp256r1" // security.AlgorithmECP256R1
+	case ocrypto.ECCModeSecp384r1:
+		req.Algorithm = "ec:secp384r1" // security.AlgorithmECP384R1
+	case ocrypto.ECCModeSecp521r1:
+		req.Algorithm = "ec:secp521r1" // security.AlgorithmECP521R1
+	case ocrypto.ECCModeSecp256k1:
+		req.Algorithm = "ec:secp256k1" // not supported
+	}
```

Comment:
Needs a `default` case that throws an error

Reply:
updated